src/Bootstrap/src/Autoconfig/HostBuilderExtensions.cs

```diff
@@ -0,0 +1,245 @@
+                hostBuilder.WireIfAnyLoaded(WireOracleConnection, OracleTypeLocator.Assemblies);
+                hostBuilder.WireIfAnyLoaded(WirePostgresConnection, PostgreSqlTypeLocator.Assemblies);
+                hostBuilder.WireIfAnyLoaded(WireRabbitMqConnection, RabbitMQTypeLocator.Assemblies);
+                hostBuilder.WireIfAnyLoaded(WireRedisConnectionMultiplexer, RedisTypeLocator.StackExchangeAssemblies);
+                hostBuilder.WireIfAnyLoaded(WireDistributedRedisCache, RedisTypeLocator.MicrosoftAssemblies.Except(new[] { "Microsoft.Extensions.Caching.Abstractions" }).ToArray());
+                hostBuilder.WireIfAnyLoaded(WireSqlServerConnection, SqlServerTypeLocator.Assemblies);
+            }
+
+            hostBuilder.WireIfLoaded(WireKubernetesActuators, SteeltoeAssemblies.Steeltoe_Management_KubernetesCore);
+            hostBuilder.WireIfLoaded(WireDiscoveryClient, SteeltoeAssemblies.Steeltoe_Discovery_ClientCore);
+            hostBuilder.WireIfLoaded(WireCloudFoundryActuators, SteeltoeAssemblies.Steeltoe_Management_CloudFoundryCore);
+            hostBuilder.WireIfLoaded(WireDistributedTracing, SteeltoeAssemblies.Steeltoe_Management_TracingCore);
+
+            if (!hostBuilder.WireIfLoaded(WireDynamicSerilog, SteeltoeAssemblies.Steeltoe_Extensions_Logging_DynamicSerilogCore))
+            {
+                hostBuilder.WireIfLoaded(WireDynamicLogging, SteeltoeAssemblies.Steeltoe_Extensions_Logging_DynamicLogger);
+            }
+
+            hostBuilder.WireIfLoaded(WireCloudFoundryContainerIdentity, SteeltoeAssemblies.Steeltoe_Security_Authentication_CloudFoundryCore);
+            return hostBuilder;
+        }
+
+        private static bool IsAssemblyLoaded(string typeName)
+        {
+            try
+            {
+                Assembly.Load(typeName);
+                return true;
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+        }
+
+        private static bool WireIfLoaded(this IHostBuilder hostBuilder, Action<IHostBuilder> action, params string[] assembly)
+        {
+            if (assembly.All(IsAssemblyLoaded))
+            {
+                action(hostBuilder);
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool WireIfAnyLoaded(this IHostBuilder hostBuilder, Action<IHostBuilder> action, params string[] assembly)
+        {
+            if (assembly.Any(IsAssemblyLoaded))
+            {
+                action(hostBuilder);
+                return true;
+            }
+
+            return false;
+        }
+
+        // ReSharper disable once S1144
+        // ReSharper disable once UnusedMember.Local
+        private static bool IsRunningInCloudFoundry() => Environment.GetEnvironmentVariable("VCAP_APPLICATION") != null;
+
+        private static bool IsRunningInKubernetes() => Environment.GetEnvironmentVariable("KUBERNETES_SERVICE_HOST") != null;
```

Comment:
`Platform.IsCloudFoundry` and `Platform.IsKubernetes` are already defined in [`Steeltoe.Common.Platform`](https://github.com/SteeltoeOSS/Steeltoe/blob/main/src/Common/src/Common/Platform.cs)